reVX/least_cost_xmission/costs/cost_combiner.py

```diff
@@ -0,0 +1,141 @@
+        Load legacy dry costs from H5 and reproject if necessary
+
+        Parameters
+        ----------
+        h5_fpath : path-like
+            H5 file with dry costs
+        layer_name : str
+            Name of costs layer
+
+        Returns
+        -------
+        array-like
+            Array of costs
+        """
+        if not Path(h5_fpath).exists():
+            raise IOError(f'H5 file {h5_fpath} does not exist')
+
+        logger.debug(f'Loading dry costs layer {layer_name} from {h5_fpath}')
+        costs = self._io_handler.load_h5_layer(layer_name, h5_fpath)
+        if costs.shape == self._io_handler.shape:
+            return costs
+
+        # Dry costs have a different shape. Attempt to reproject
+        logger.debug('Dry costs shape does not match template raster. '
+                     'Reprojecting.')
+
+        attrs = self._io_handler.load_h5_attrs(layer_name, h5_fpath)
+        json_profile = attrs['profile']
+        profile = json.loads(json_profile)
+        reprojected = self._io_handler.reproject(costs, profile, init_dest=-1)
+
+        return reprojected
+
+    def combine_costs(self, wet_costs: npt.NDArray,
+                      dry_costs: npt.NDArray, landfall_cost: float,
+                      layer_name: str, save_tiff: bool = True):
+        """
+        Combine wet, dry, and landfall costs using appropriate masks
+
+        Parameters
+        ----------
+        wet_costs : array-like
+            Wet costs array
+        dry_costs : array-like
+            Dry costs array
+        landfall_cost : float
+            Cost to apply to landfall cells for conversion from underwater
+            cables to land based transmission.
+        layer_name : str
+            Layer name for combined costs in H5
+        save_tiff : bool, optional
+            Save combined costs to GeoTIFF if True, by default True
+        """
+        if wet_costs.shape != self._io_handler.shape:
+            raise ValueError(f'Wet costs shape {wet_costs.shape} does not '
+                             'match shape of template raster '
+                             f'{self._io_handler.shape}')
+
+        if dry_costs.shape != self._io_handler.shape:
+            raise ValueError(f'Dry costs shape {dry_costs.shape} does not '
+                             'match shape of template raster '
+                             f'{self._io_handler.shape}')
+
+        combined = np.zeros(self._io_handler.shape, dtype=DEFAULT_DTYPE)
+        combined[self._masks.wet_mask] = wet_costs[self._masks.wet_mask]
+        combined[self._masks.dry_mask] = dry_costs[self._masks.dry_mask]
+
+        # TODO - include half of wet and half of dry cost in landfall cost
+        combined[self._masks.landfall_mask] = landfall_cost
+
+        if save_tiff:
+            logger.debug('Saving combined costs to GeoTIFF')
+            self._io_handler.save_tiff(combined, COMBINED_COSTS_TIFF)
+
+        num_zeros = (combined == 0).sum()
+        if num_zeros > 0:
```

Comment:
Every cell should have a cost, and a zero is likely a sign of some error in the cost creation process. 

The MCP module will not route though any cell that is negative. We probably shouldn't be using negatives. However, currently the dry costs use -1 (or -10?) for anything out side of the defined ISO regions (see red in the image below). I'm not sure how best to go about this. 
![image](https://github.com/NREL/reVX/assets/18619942/e47f7f85-9c24-44c8-bf4f-9f43cad674fd)

Reply:
Well, it does make sense to have some value indicate that a cell simply can never be routed though, no matter what. I just couldn't remember what that convention was. I think negative values make sense to me, in which case I would agree that zero's in the cost array could be problematic. 

Do you think it would be worthwhile to mention in the warning message that a user can use negative values to specify NO-ROUTE cells? I think that was my biggest confusion seeing the message about 0's at a glance

Reply:
**TL;DR: I vote we revisit this when I add the dry costs to the new CLI**

A couple thoughts:

1. The current additive costs/friction/barrier system will cancel out a negative cost anywhere a friction or barrier overlaps it. It still has a high routing cost, but a negative actual cost, which doesn't affect routing. So kinda weird. Transferring negative costs to the routing layer wouldn't be that hard though. 
2. We experimented with using -1 for barriers in CONUS. It didn't work, but that was in part because we only had one barrier layer. We might be able to revisit that and have certain layers that are negative, and others that are just normal barriers, but that is adding additional complexity that I'm not sure we really need. 
3. The only real negatives right now are in the dry costs, and I'm not sure what the options there are. I vote we revisit this when I add the dry costs to the new CLI. 
4. This possibly should move into a bigger conversation regarding QA/QC for the routing layers and the transmission paths. During the development of the CONUS code, Anthony, myself, etc spent a decent amount of time staring at the generated paths and verifying they were sane. A lot weren't, and we had to tweak stuff. Analysts shouldn't be using costs without verifying transmission path alignments first. Forgetting something like the barriers in Mexico, or not setting a landfall cost for a CONUS analysis could easily result in paths cheating and cutting through Mexico or though the ocean.

Reply:
Yes, ok. Let's revisit with Dry costs